pkg/gzip/gzip.go

```diff
@@ -56,6 +57,12 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 		}
 
 		parent := filepath.Dir(fpath)
+
+		// Ignore all paths that would end up outside the destination directory.
+		if !strings.HasPrefix(parent, dest) {
+			continue
```

Comment:
My own ranked preferences are:
1. Call the archive "invalid" and reject it
2. Extract into the destination dir: `/my/jail` + `../../bin/bash` => `/my/jail/bin/bash`
3. Ignore the file
4. Silently ignore file  <= this will bite us eventually